SQL_Assignments/sql_2.sql

```diff
@@ -0,0 +1,137 @@
+Create a table students and insert names in malayalam
+
+    create table employee(empid int primary key,empname nvarchar(20))
+    insert into employee values(1,N'അഭിഷേക്')
+
+Retrieve all employees who work in Sales, Marketing, or IT departments.
+
+
+    select Firstname,lastname from emp where Department in('IT','managment','sales')
+
+
+Find all employees with salaries ranging from $50,000 to $75,000 (inclusive).
+
+
+    select firstname,lastname from emp where Salary between 50000 and 75000
+
+
+List all employees whose last name begins with the letter 'S'.
+
+    select firstname,lastname from emp where lastname like 's%'
+
+Display all employees with exactly five letters in their first name.
+
+    select firstname,lastname from emp where lastname like '_____'
+
+Find employees whose last name starts with either 'B', 'R', or 'S'.
+
+    select firstname,lastname from emp where lastname like 'B%' or lastname like 'S%' or lastname like 'R%'
+
+Retrieve all employees whose first name begins with any letter from 'A' through 'M'.
+
+    select firstname,lastname from emp where firstname like '[a-m]%'
+
+List employees whose last name doesnt start with a vowel (A, E, I, O, U).
+
+    select firstname,lastname from emp where lastname not like 'a%' and lastname not like 'e%' and lastname not like 'i%' and lastname not like 'o%' and lastname not like 'u%'
+
+
+
+Identify employees earning more than $80,000 annually. 
+
+    select firstname,lastname from emp where Salary>80000
+
+
+Find employees who joined the company before 2020. 
+
+    select firstname,lastname from emp where HireDate<'01-01-2020'
+
+ List all employees not named 'John' (first name).
+
+    select firstname,lastname from emp where FirstName not like 'john'
+Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+
+    select firstname,lastname from emp  where Department ='marketing' and Salary <70000 and HireDate> '06-30-2019'
+
+Find employees whose first name contains the letters 'an' anywhere and ends with 'e'.
+
+    select firstname,lastname from emp  where FirstName like '%an%e'
+
+
+
+
+
+
+
+
+CREATE TABLE Sales (
+    SaleID INT PRIMARY KEY,
+    ProductID INT,
+    CustomerID INT,
+    SaleDate DATE,
+    Quantity INT,
+    UnitPrice DECIMAL(10, 2)
```

Comment:
Why do we need to create decimal columns like this ?

Reply:
for more precise price .00 is given as decimal(10,2)